modules/nf-scil/reconst/frf/main.nf

```diff
@@ -24,40 +27,91 @@ process RECONST_FRF {
     def fa_min = task.ext.fa_min ? "--min_fa " + task.ext.fa_min : ""
     def nvox_min = task.ext.nvox_min ? "--min_nvox " + task.ext.nvox_min : ""
     def roi_radius = task.ext.roi_radius ? "--roi_radii " + task.ext.roi_radius : ""
+    def dwi_shell_tolerance = task.ext.dwi_shell_tolerance ? "--tolerance " + task.ext.dwi_shell_tolerance : ""
+    def max_dti_shell_value = task.ext.max_dti_shell_value ?: 1500
+    def b0_thr_extract_b0 = task.ext.b0_thr_extract_b0 ?: 10
+    def dti_shells = task.ext.dti_shells ?: "\$(cut -d ' ' --output-delimiter=\$'\\n' -f 1- $bval | awk -F' ' '{v=int(\$1)}{if(v<=$max_dti_shell_value|| v<=$b0_thr_extract_b0)print v}' | uniq)"
+    def set_method = task.ext.method ? task.ext.method : "ssst"
+
+    def fa_thr_wm = task.ext.fa_thr_wm ? "--fa_thr_wm " + task.ext.fa_thr_wm : ""
+    def fa_thr_gm = task.ext.fa_thr_gm ? "--fa_thr_gm " + task.ext.fa_thr_gm : ""
+    def fa_thr_csf = task.ext.fa_thr_csf ? "--fa_thr_csf " + task.ext.fa_thr_csf : ""
+    def md_thr_wm = task.ext.md_thr_wm ? "--md_thr_wm " + task.ext.md_thr_wm : ""
+    def md_thr_gm = task.ext.md_thr_gm ? "--md_thr_gm " + task.ext.md_thr_gm : ""
+    def md_thr_csf = task.ext.md_thr_csf ? "--md_thr_csf " + task.ext.md_thr_csf : ""
+
     def fix_frf = task.ext.manual_frf ? task.ext.manual_frf : ""
+    def fix_wm_frf = task.ext.manual_wm_frf ? task.ext.manual_wm_frf : ""
+    def fix_gm_frf = task.ext.manual_gm_frf ? task.ext.manual_gm_frf : ""
+    def fix_csf_frf = task.ext.manual_csf_frf ? task.ext.manual_csf_frf : ""
     def set_mask = mask ? "--mask $mask" : ""
+    def set_wm_mask = wm_mask ? "--mask_wm $wm_mask" : ""
+    def set_gm_mask = gm_mask ? "--mask_gm $gm_mask" : ""
+    def set_csf_mask = csf_mask ? "--mask_csf $csf_mask" : ""
 
     """
     export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
     export OMP_NUM_THREADS=1
     export OPENBLAS_NUM_THREADS=1
 
-    scil_compute_ssst_frf.py $dwi $bval $bvec ${prefix}__frf.txt \
-        $set_mask $fa $fa_min $nvox_min $roi_radius --force_b0_threshold
+    if [ "$set_method" = "ssst" ]
+    then
+
+        scil_dwi_extract_shell.py $dwi $bval $bvec $dti_shells \
+                dwi_dti_shells.nii.gz bval_dti_shells bvec_dti_shells \
+                $dwi_shell_tolerance -f
```

Comment:
Same comment

Reply:
This is tricky, since it should be the exact same shells that are selected for both FRF and FODF modules (in the case of msmt). Normally, I simply give a limit for the DTI b-values (for computing the diffusivities for FRF), and the script uses all the shells after. Here, we would need a new parameter, like selected_shells, for msmt only.

For the ssst, the extract shells step is there to ensure we use only DTI shells. This is done inside the msmt script (python) with the max_dti_shell_value parameter.

If I understand correctly, your goal is to select some shells for the FODF reconstruction?